flake.nix

```diff
@@ -0,0 +1,68 @@
+{
+  inputs = {
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    utils.url = "github:numtide/flake-utils";
+  };
+  outputs = { self, nixpkgs, utils, ... }@inputs: utils.lib.eachDefaultSystem (system:
+    let
+      # toml info
+      cargoTOML = (builtins.fromTOML (builtins.readFile ./Cargo.toml));
+      packageVersion = cargoTOML.workspace.package.version;
+      rustVersion = cargoTOML.workspace.package.rust-version;
+      rustPkg = pkgs.rust-bin.stable."${rustVersion}".default.override {
+        extensions = [ "rust-src" "rust-analyzer" "clippy" "rustfmt" ];
+      };
+      # platform packages
+      macPackages = pkgs.lib.optionals pkgs.stdenv.isDarwin (with pkgs.darwin.apple_sdk.frameworks; [ Security CoreFoundation CoreServices ]);
+      linuxPackages = pkgs.lib.optionals pkgs.stdenv.isLinux (with pkgs; [
+        libclang.lib
+        llvmPackages.libcxxClang
+        clang
+        openssl
```

Comment:
Nice catch, openssl isn't needed. I added it for testing since I saw some the workspaces were network related and was curious if they would link to openssl as I have seen other Rust packages do. I updated the pr to remove that.